### ISSUE TITLE
Why Resource Class is not being included in new CosmosDB V3 SDK, which was there in old DocumentDB v2 SDK?

### ISSUE THREAD

**Issue by sumitshrm4** [Azure/azure-cosmos-dotnet-v3#1694]
We are upgrading our code from old DocumentDB V2 SDK to new V3 SDK. We had used this Resource as a base class to all the models which we are saving in cosmos db because of their automatic Id generation as well as Timestamp properties.

I could not find Resource Class which was part of old DocumentDB SDK in the new SDK V3. I could only find 2 similar classes Database properties class Container properties class. But I am not sure if i should use them or create my own base class.

I know I can create my own base class, but i just don't want to do that unless it is already not available in the SDK.

**Comment by j82w** [comment]
TLDR: v3 does not have the equivalent Resource as a base class.

The reasons why:
1. Auto id generation was causing a lot of confusion, and often caused customers to design inefficient workflows. For example there was several customers doing a query later to find the item instead of just doing a point read. Doing a query is far more expensive and slower compared to a point reads. By the time customers realized this there application was in production and causing it to be far more painful process to fix.
2. In v3 SDK it allows custom serializer, so you could use the new System.Text.Json library. Having a base Resource class would make it dependent on the current Newtonsoft default.
3. This also provides a better ownership model as you don't have to change your models in the future when you move to the v4 SDK when it is released. 
4. We also found most users did not need all the properties or were not using the class at all. It was a simple enough class that any user that did need it could easy create their own without taking a dependency on the SDK.

**Comment by sumitshrm4** [comment]
Thanks for clarification!!
